miso-web/src/main/webapp/scripts/hot_sample.js

```diff
@@ -675,6 +687,84 @@ HotTarget.sample = (function() {
           }
         
         },
+        Constants.isDetailedSample
+            ? {
+              name : "Edit Parents",
+              action : function(samples) {
+                
+                // Construct a graph of the sample valid relationships and mark
+                // each node with what samples descend from that node
+                var graph = {};
+                var buildGraph = function(sampleId, sampleClass) {
+                  var node;
+                  if (graph[sampleClass.id]) {
+                    node = graph[sampleClass.id];
+                    var queue = [].concat(node.parents);
+                    while (queue.length > 0) {
+                      var parentNode = queue.pop();
+                      queue = queue.concat(parentNode.parents);
+                      parentNode.seen[sampleId] = true;
+                    }
+                  } else {
+                    node = {
+                      sampleClass : sampleClass,
+                      seen : {},
+                      parents : Constants.sampleValidRelationships
+                          .filter(
+                              function(relation) {
+                                return relation.childId == sampleClass.id && relation.parentId != sampleClass.id;
+                              }).map(
+                              function(relation) {
+                                return buildGraph(sampleId, Utils.array
+                                    .findFirstOrNull(function(sc) {
+                                      return sc.id == relation.parentId;
+                                    }, Constants.sampleClasses));
+                              })
+                    
+                    };
+                    graph[sampleClass.id] = node;
+                  }
+                  node.seen[sampleId] = true;
+                  return node;
+                }
+                samples.forEach(function(sample) {
+                  buildGraph(sample.id, Utils.array.findFirstOrNull(
+                      function(sc) {
+                        return sc.id == sample.sampleClassId;
+                      }, Constants.sampleClasses));
+                });
+                // Traverse the graph for any sample class that has been visited
+                // by every sample.
+                var allowedParents = Constants.sampleClasses.filter(
+                    function(sampleClass) {
+                      var node = graph[sampleClass.id];
+                      return node && samples.every(function(sample) {
+                        return node.seen[sample.id];
+                      });
+                    }).sort(Utils.sorting.sampleClassComparator);
+                
+                if (allowedParents.length == 0) {
+                  Utils.showOkDialog("Edit Parents",
+                      [ "No common parents to edit." ]);
+                  return;
+                  return;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this code after the "return" statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AUnreachableCode)